Create supported-os.md for .NET 9 (#9128)
dotnet/core: release-notes/9.0/supported-os.md

# .NET 9 - Supported OS versions

> [.NET 9](README.md) is a [Standard Term Support (STS)](../../release-policies.md) release and [is supported](../../microsoft-support.md) on multiple operating systems per their lifecycle policy.

Reviewer: ```suggestion
[.NET 9](README.md) is a [Standard Term Support (STS)](../../release-policies.md) release and [is supported](../../support.md) on multiple operating systems per their lifecycle policy.
```

For issues with .NET on operating systems not listed here, please open a GitHub issue in the appropriate .NET repository or contact the OS maintainer community. See [.NET Repos](../../Documentation/core-repos.md) for the repository list.

## Windows

OS                                    | Version                 | Architectures   | Lifecycle
--------------------------------------|-------------------------|-----------------|----------
[Windows 10 Client][Windows-client]   | Version 1607+           | x64, x86, Arm64 | [Windows][Windows-lifecycle]
[Windows 11][Windows-client]          | Version 22000+          | x64, x86, Arm64 | [Windows][Windows-lifecycle]
[Windows Server][Windows-Server]      | 2012+                   | x64, x86        | [Windows Server][Windows-Server-lifecycle]
[Windows Server Core][Windows-Server] | 2012+                   | x64, x86        | [Windows Server][Windows-Server-lifecycle]
[Nano Server][Nano-Server]            | Version 1809+           | x64             | [Windows Server][Windows-Server-lifecycle]

> [Windows-client]: https://www.microsoft.com/windows/

Reviewer: Is there a better link than this? It is very salesy.

Reviewer 2: there's an MS Learn one similar to what we use for Windows Server which looks better to me and doesn't fail the markdownlint check: https://github.com/dotnet/core/pull/9145

--------------------------------------|-----------------------|-------------------|----------
[Alpine Linux][Alpine]                | 3.17+                 | x64, Arm64, Arm32 | [Alpine][Alpine-lifecycle]
[Debian][Debian]                      | 11+                   | x64, Arm64, Arm32 | [Debian][Debian-lifecycle]
[Fedora][Fedora]                      | 38+                   | x64               | [Fedora][Fedora-lifecycle]
[openSUSE][OpenSUSE]                  | 15+                   | x64               | [OpenSUSE][OpenSUSE-lifecycle]
[Oracle Linux][Oracle-Linux]          | 8+                    | x64               | [Oracle][Oracle-lifecycle]
[Red Hat Enterprise Linux][RHEL]      | 8+                    | x64, Arm64        | [Red Hat][RHEL-lifecycle]
[SUSE Enterprise Linux (SLES)][SLES]  | 12 SP5+               | x64               | [SUSE][SLES-lifecycle]
[Ubuntu][Ubuntu]                      | 20.04+                | x64, Arm64, Arm32 | [Ubuntu][Ubuntu-lifecycle]

Other distributions are supported at best effort, per [.NET Support and Compatibility for Linux Distributions](../../linux-support.md).

### Libc compatibility

> - [glibc][glibc] 2.23 (from Ubuntu 16.04)

Reviewer 3: The discussion in https://github.com/dotnet/runtime/issues/91826 suggested to bump this to 18.04. Is it intentional that we are keeping this at 16.04?

cc @richlander

Reviewer: You are right. 18.04.

Reviewer: ```suggestion
- [glibc][glibc] 2.27 (from Ubuntu 18.04)
```

# .NET 9 - Supported OS versions

> [.NET 9](README.md) is a [Long Term Support (LTS)](../../release-policies.md) release and [is supported](../../microsoft-support.md) on multiple operating systems per their lifecycle policy.

Reviewer 4: I thought even number releases were LTS only. Was this recently changed?

Reviewer: No. That's a copy/paste error.

```suggestion
[.NET 9](README.md) is a [Standard Term Support (STS)](../../release-policies.md) release and [is supported](../../microsoft-support.md) on multiple operating systems per their lifecycle policy.
```

Author: Updated the doc with "STS" in-line with our support policy

> For issues with .NET on operating systems not listed here, please open a GitHub issue in the appropriate .NET repository or contact the OS maintainer community . See [.NET Repos](../../Documentation/core-repos.md) for the repository list.

Reviewer 4: ```suggestion
For issues with .NET on operating systems not listed here, please open a GitHub issue in the appropriate .NET repository or contact the OS maintainer community. See [.NET Repos](../../Documentation/core-repos.md) for the repository list.
```

--------------------------------------|-------------------------|-----------------|----------
[Windows 10 Client][Windows-client]   | Version 1607+           | x64, x86, Arm64 | [Windows][Windows-lifecycle]
[Windows 11][Windows-client]          | Version 22000+          | x64, x86, Arm64 | [Windows][Windows-lifecycle]
[Windows Server][Windows-Server]      | 2012+                   | x64, x86        | [Windows Server][Windows-Server-lifecycle]
[Windows Server Core][Windows-Server] | 2012+                   | x64, x86        | [Windows Server][Windows-Server-lifecycle]
[Nano Server][Nano-Server]            | Version 1809+           | x64             | [Windows Server][Windows-Server-lifecycle]

[Windows-client]: https://www.microsoft.com/windows/
[Windows-lifecycle]: https://support.microsoft.com/help/13853/windows-lifecycle-fact-sheet
[win-client-docker]: https://hub.docker.com/_/microsoft-windows
[Windows-Server-lifecycle]: https://learn.microsoft.com/windows-server/get-started/windows-server-release-info
[Nano-Server]: https://learn.microsoft.com/windows-server/get-started/getting-started-with-nano-server
[Windows-Server]: https://learn.microsoft.com/windows-server/

> .NET 8 is supported in the x64 emulator on Windows 11 Arm64.

Reviewer 5: 9, presumably?

Reviewer: ```suggestion
.NET 9 is supported in the x64 emulator on Windows 11 Arm64.
```

[Oracle-Lifecycle]: https://www.oracle.com/a/ocom/docs/elsp-lifetime-069338.pdf
[RHEL]: https://www.redhat.com/en/technologies/linux-platforms/enterprise-linux
[RHEL-lifecycle]: https://access.redhat.com/support/policy/updates/errata/
[RHEL-msft-pm]: https://learn.microsoft.com/dotnet/core/install/linux-package-manager-rhel8
[RHEL-pm]: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/developing_.net_applications_in_rhel_8/using-net-core-on-rhel_gsg#installing-net-core_gsg
[SLES]: https://www.suse.com/products/server/
[SLES-lifecycle]: https://www.suse.com/lifecycle/
[SLES-pm]: https://learn.microsoft.com/dotnet/core/install/linux-package-manager-sles15
[Ubuntu]: https://ubuntu.com/
[Ubuntu-lifecycle]: https://wiki.ubuntu.com/Releases
[Ubuntu-pm]: https://learn.microsoft.com/dotnet/core/install/linux-package-manager-ubuntu-2004
[glibc]: https://www.gnu.org/software/libc/
[musl]: https://musl.libc.org/

## macOS

OS                            | Version                   | Architectures     |
------------------------------|---------------------------|-------------------|
[macOS][macOS]                | 12.0+                     | x64, Arm64        |

> .NET 8 is supported in the Rosetta 2 x64 emulator.

Reviewer 5: 9?

Reviewer: ```suggestion
.NET 9 is supported in the Rosetta 2 x64 emulator.
```

[macOS]: https://support.apple.com/macos

## Android

OS                            | Version                 | Architectures     |
------------------------------|-------------------------|-------------------|
[Android][Android]            | API 21+                 | x64, Arm32, Arm64 |

[Android]: https://support.google.com/android

## iOS / tvOS / MacCatalyst

OS                            | Version                 | Architectures     |
------------------------------|-------------------------|-------------------|
[iOS][iOS]                    | 12.0+                   | Arm64             |
[iOS Simulator][iOS]          | 12.0+                   | x64, Arm64        |
[tvOS][tvOS]                  | 12.0+                   | Arm64             |
[tvOS Simulator][tvOS]        | 12.0+                   | x64, Arm64        |
[MacCatalyst][macOS]          | 12.0+                   | x64, Arm64        |

[iOS]: https://support.apple.com/ios
[tvOS]: https://support.apple.com/apple-tv

## QEMU

> .NET 8 is not supported being run (emulated) via [QEMU](https://www.qemu.org/). QEMU is used, for example, to emulate Arm64 containers on x64, and vice versa.

Reviewer: ```suggestion
.NET 9 is not supported being run (emulated) via [QEMU](https://www.qemu.org/). QEMU is used, for example, to emulate Arm64 containers on x64, and vice versa.
```

## Support changes from .NET 6.0

> The following operating systems are no longer supported, starting with .NET 7.0.

Reviewer 5: The section says 6, the words say 7; I suspect they both mean 8?

OS                                    | Version                 | Architectures     |
--------------------------------------|-------------------------|-------------------|

> [Windows Client][Windows-client]      | 7 SP1, 8.1      | x64, x86          |

Reviewer 4: There is a bit more nuance to this if I recall. @richlander @jkotas Is this how we want to express Windows 7 support?

Reviewer: I added a suggestion to remove that.

> ## Support changes from .NET 6.0
> 
> The following operating systems are no longer supported, starting with .NET 7.0.
> 
> OS                                    | Version                 | Architectures     |
> --------------------------------------|-------------------------|-------------------|
> [Windows Client][Windows-client]      | 7 SP1, 8.1      | x64, x86          |

Reviewer: I don't think we need this content. Also, the references to .NET 6 and 7 are confusing.

```suggestion
```
